skglm/estimators.py

```diff
@@ -1675,3 +1691,201 @@ def fit(self, X, y):
+                        W_11[:, di] = W[:, di][indices != col]
+                    else:
+                        W_11[:] = W[1:, 1:]
+
+                s_12 = S[col, indices != col]
+
+                if self.algo == "dual":
+                    beta_init = (Theta[indices != col, col] /
+                                 (Theta[col, col] + 1000 * eps))
+                    Q = W_11
+
+                elif self.algo == "primal":
+                    inv_Theta_11 = (W[_11] -
+                                    np.outer(W[_12],
+                                             W[_12])/W[_22])
+                    Q = inv_Theta_11
+                    beta_init = Theta[indices != col, col] * S[col, col]
+                else:
+                    raise ValueError(f"Unsupported algo {self.algo}")
+
+                beta = barebones_cd_gram(
+                    Q,
+                    s_12,
+                    x=beta_init,
+                    alpha=self.alpha,
+                    weights=Weights[indices != col, col],
+                    tol=self.inner_tol,
+                    max_iter=self.max_iter,
+                )
+
+                if self.algo == "dual":
+                    w_12 = -np.dot(W_11, beta)
+                    W[col, indices != col] = w_12
+                    W[indices != col, col] = w_12
+
+                    Theta[col, col] = 1 / \
+                        (W[col, col] + np.dot(beta, w_12))
+                    Theta[indices != col, col] = beta*Theta[col, col]
+                    Theta[col, indices != col] = beta*Theta[col, col]
+
+                else:  # primal
+                    Theta[indices != col, col] = beta / S[col, col]
+                    Theta[col, indices != col] = beta / S[col, col]
+                    Theta[col, col] = (1/S[col, col] +
+                                       Theta[col, indices != col] @
+                                       inv_Theta_11 @
+                                       Theta[indices != col, col])
+                    W[col, col] = (1/(Theta[col, col] -
+                                      Theta[indices != col, col] @
+                                      inv_Theta_11 @
+                                      Theta[indices != col, col]))
+                    W[indices != col, col] = (-W[col, col] *
+                                              inv_Theta_11 @
+                                              Theta[indices != col, col])
+                    W[col, indices != col] = (-W[col, col] *
+                                              inv_Theta_11 @
+                                              Theta[indices != col, col])
+                    # Maybe W_11 can be done smarter ?
+                    W[_11] = (inv_Theta_11 +
+                              np.outer(W[indices != col, col],
+                                       W[indices != col, col])/W[col, col])
+
+            if np.linalg.norm(Theta - Theta_old) < self.tol:
```

Comment:
add a verbose parameter, and print the msg only in this case : we don't want to print stuff everytime we call fit